src/lib/router.ts

```diff
@@ -100,50 +105,59 @@ export function updateProjectRoute(
   }
 }
 
-export const push = (newRoute: Route): void => {
-  const history = get(historyStore);
+const loadExecutor = mutexExecutor.create();
+
+async function mutexLoad(newRoute: Route): Promise<void> {
+  isLoading.set(true);
+
+  const loadedRoute = await loadExecutor.run(async () => {
+    return loadRoute(newRoute);
+  });
+
+  // Only let the last request through.
+  if (loadedRoute === undefined) {
+    return;
+  }
+
+  activeRouteStore.set(loadedRoute);
+  isLoading.set(false);
+}
 
-  // Limit history to a maximum of 10 steps. We shouldn't be doing more than
-  // one subsequent pop() anyway.
-  historyStore.set([...history, newRoute].slice(-10));
+export async function push(newRoute: Route): Promise<void> {
+  await mutexLoad(newRoute);
 
   const path = import.meta.env.VITE_HASH_ROUTING
     ? "#" + routeToPath(newRoute)
     : routeToPath(newRoute);
 
-  window.history.pushState(newRoute, documentTitle, path);
-};
+  window.history.pushState(newRoute, DOCUMENT_TITLE, path);
+}
 
-export const pop = (): void => {
-  const history = get(historyStore);
-  const newRoute = history.pop();
-  if (newRoute) {
-    historyStore.set(history);
-    window.history.back();
-  }
-};
+export function pop(): void {
+  window.history.back();
+}
```

Comment:
Do we really need this function or can we just inline it in the two places it is used?

If you want to keep this function I would also consider renaming this to `back()`. Consistent names make it easier to understand the code.

Reply:
I guess we don't need it: 4c2d8acbae874e4f6c2b1e1d61425f92ac57943b. I think my initial motivation was to show people who don't know how the router works that you use `window.history.back()` for backwards navigation. I assumed that people would look for it in the router. 🤷